From b9cdb540402c74de35c4894f6b5c728e3606be08 Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Mon, 3 Nov 2025 19:30:04 +0200
Subject: [PATCH 1/1] platform/marvell-prest: fdt_cn9131 link to
 usr/lib/linux-image-arm64

Use fdt_cn9131 pointing to link usr/lib/linux-image-arm64/ just like it
is used by all other platforms (instead of /boot/ used in Bookworm)

Signed-off-by: Yan Markman <ymarkman@marvell.com>
---
 platform/marvell-prestera/sonic_fit.its | 87 +++++++++++++------------
 1 file changed, 46 insertions(+), 41 deletions(-)

diff --git a/platform/marvell-prestera/sonic_fit.its b/platform/marvell-prestera/sonic_fit.its
index 4beb22566..59f10dbfc 100644
--- a/platform/marvell-prestera/sonic_fit.its
+++ b/platform/marvell-prestera/sonic_fit.its
@@ -18,28 +18,33 @@
                 algo = "sha1";
             };
         };
-        fdt_ac5x {
-            description = "Flattened Device Tree blob for AC5x";
-            data = /incbin/("/usr/lib/linux-image-6.12.41+deb13-sonic-arm64/marvell/ac5-98dx35xx-rd.dtb");
-            type = "flat_dt";
+        kernel_cn9131 {
+            description = "Linux Kernel - cn9131";
+            data = /incbin/("/boot/vmlinuz-6.12.41+deb13-sonic-arm64");
+            type = "kernel";
             arch = "arm64";
+            os = "linux";
             compression = "none";
-            load = <0x2 0x1000000>;
+            load = <0x1000000>;
+            entry = <0x1000000>;
             hash@1 {
                 algo = "sha1";
             };
         };
-        fdt_7215_a1 {
-            description = "Flattened Device Tree blob for 7215_IXS_A1";
-            data = /incbin/("/usr/lib/linux-image-6.12.41+deb13-sonic-arm64/marvell/7215-ixs-a1.dtb");
-            type = "flat_dt";
+        kernel_7215_a1 {
+            description = "Linux Kernel - 7215_a1/Nokia";
+            data = /incbin/("/boot/vmlinuz-6.12.41+deb13-sonic-arm64");
+            type = "kernel";
             arch = "arm64";
+            os = "linux";
             compression = "none";
-            load = <0x11000000>;
+            load = <0x12000000>;
+            entry = <0x12000000>;
             hash@1 {
                 algo = "sha1";
             };
         };
+
         ramdisk_ac5x {
             description = "ramdisk for AC5x";
             data = /incbin/("/boot/initrd.img-6.12.41+deb13-sonic-arm64");
@@ -53,61 +58,59 @@
                 algo = "sha1";
             };
         };
-        kernel_1 {
-            description = "Linux Kernel";
-            data = /incbin/("/boot/vmlinuz-6.12.41+deb13-sonic-arm64");
-            type = "kernel";
+        ramdisk_cn9131 {
+            description = "ramdisk - cn9131";
+            data = /incbin/("/boot/initrd.img-6.12.41+deb13-sonic-arm64");
+            type = "ramdisk";
             arch = "arm64";
             os = "linux";
             compression = "none";
-            load = <0x1000000>;
-            entry = <0x1000000>;
+            load = <0x6000000>;
+            entry = <0x6000000>;
             hash@1 {
                 algo = "sha1";
             };
         };
-        ramdisk_1 {
-            description = "ramdisk";
+        ramdisk_7215_a1 {
+            description = "ramdisk - 7215_a1/Nokia";
             data = /incbin/("/boot/initrd.img-6.12.41+deb13-sonic-arm64");
             type = "ramdisk";
             arch = "arm64";
             os = "linux";
             compression = "none";
-            load = <0x6000000>;
-            entry = <0x6000000>;
+            load = <0x18000000>;
+            entry = <0x18000000>;
             hash@1 {
                 algo = "sha1";
             };
         };
-        kernel_2 {
-            description = "Linux Kernel";
-            data = /incbin/("/boot/vmlinuz-6.12.41+deb13-sonic-arm64");
-            type = "kernel";
+
+        /* Take care to sync sync with platform_arm64.conf */
+        fdt_ac5x {
+            description = "Flattened Device Tree blob for AC5x";
+            data = /incbin/("/usr/lib/linux-image-6.12.41+deb13-sonic-arm64/marvell/ac5-98dx35xx-rd.dtb");
+            type = "flat_dt";
             arch = "arm64";
-            os = "linux";
             compression = "none";
-            load = <0x12000000>;
-            entry = <0x12000000>;
+            load = <0x2 0x1000000>;
             hash@1 {
                 algo = "sha1";
             };
         };
-        ramdisk_2 {
-            description = "ramdisk";
-            data = /incbin/("/boot/initrd.img-6.12.41+deb13-sonic-arm64");
-            type = "ramdisk";
+        fdt_7215_a1 {
+            description = "Flattened Device Tree blob for 7215_IXS_A1";
+            data = /incbin/("/usr/lib/linux-image-6.12.41+deb13-sonic-arm64/marvell/7215-ixs-a1.dtb");
+            type = "flat_dt";
             arch = "arm64";
-            os = "linux";
             compression = "none";
-            load = <0x18000000>;
-            entry = <0x18000000>;
+            load = <0x11000000>;
             hash@1 {
                 algo = "sha1";
             };
         };
         fdt_cn9131 {
-            description = "Flattened Device Tree blob";
-            data = /incbin/("/boot/cn9131-db-comexpress.dtb");
+            description = "Flattened Device Tree blob for cn9131";
+            data = /incbin/("/usr/lib/linux-image-6.12.41+deb13-sonic-arm64/marvell/cn9131-db-comexpress.dtb");
             type = "flat_dt";
             arch = "arm64";
             compression = "none";
@@ -118,31 +121,33 @@
             };
         };
     };
+
+    /* From Uboot env fit_conf_name=#conf_cn9131 */
     configurations {
         default = "conf_ac5x";
         conf_ac5x {
             description = "Boot Linux kernel with FDT blob + ramdisk for RD_AC5x";
             kernel = "kernel_ac5x";
-            fdt = "fdt_ac5x";
             ramdisk = "ramdisk_ac5x";
+            fdt = "fdt_ac5x";
             hash@1 {
                 algo = "sha1";
             };
         };
         conf_7215_a1 {
             description = "Boot Linux kernel with FDT blob + ramdisk for 7125_IXS_A1";
-            kernel = "kernel_2";
+            kernel = "kernel_7215_a1";
+            ramdisk = "ramdisk_7215_a1";
             fdt = "fdt_7215_a1";
-            ramdisk = "ramdisk_2";
             hash@1 {
                 algo = "sha1";
             };
         };
         conf_cn9131 {
             description = "Boot Linux kernel with FDT blob + ramdisk for CN9131";
-            kernel = "kernel_1";
+            kernel = "kernel_cn9131";
+            ramdisk = "ramdisk_cn9131";
             fdt = "fdt_cn9131";
-            ramdisk = "ramdisk_1";
             hash@1 {
                 algo = "sha1";
             };
-- 
2.34.1

